### ISSUE TITLE
ESM import error

### ISSUE THREAD

**Issue by btakita** [lukeed/sirv#114]
```mjs
import * as sirv_o from 'sirv'
const { default: sirv } = sirv_o
```

Results with the error:

```
'default' is not exported by ../../node_modules/.pnpm/mime@2.5.2/node_modules/mime/lite.js, imported by ../../node_modules/.pnpm/sirv@1.0.12/node_modules/sirv/build.mjs
3: import { parse } from '@polka/url';
4: import list from 'totalist/sync';
5: import mime from 'mime/lite';
```

**Comment by lukeed** [comment]
`import * as` imports as an object.

This module does not have native ESM enabled, so unless [you're bundling](https://github.com/lukeed/sirv/blob/master/packages/sirv/package.json#L6), you'll be using the CommonJS entry.

Even so, this is the correct format. It will always worked and I just verified locally as well:

```js
import sirv from 'sirv';
```
Your TS will be pissed off, but it'll be correct.

The format you used should only be used when you have multiple named imports, or if you have a default _with_ named exports.

**Comment by btakita** [comment]
Not using TS. Thas was from a `*.mjs` file. Using rollup without `@rollup/common-js-plugin` since it does not seem to work with `@swc/core`

**Comment by lukeed** [comment]
I just reread the error message, sorry

My `mime` import is incorrect. It needs to be `import * as` while yours needs to be `import sirv` :D 

I'll fix this tomorrow morning and publish a new patch.

Just mentioned TS for potential type checking errors.

**Comment by btakita** [comment]
Thank you...I think I was having this sort of issue with a few of your packages. I'll be more diligent about error reporting if I come across any issue as I have more time now.

**Comment by lukeed** [comment]
If, however, your `sirv` import is _actually_ a `sirv/build.mjs` import, then yes, you will get errors. 

This is expected & works as (Node) designed, since `mime/lite` and `totalist/sync` do not have those "module entry files" defined as exports. While I obviously control `totalist` and could take care of that, changes to `mime` would need to make changes [to support native ESM too](https://github.com/broofa/mime/pull/233#issuecomment-892255597)

But, as seen from my above example, none of this is needed. Node ESM is able to follow the dependency resolutions correctly _even though/if_ they happen to be loading CommonJS code. If Node were not able to do this, native ESM would be dead on arrival since module interop would be impossible. (This was momentarily true, and was quickly fixed)

**Comment by lukeed** [comment]
The mime issue is resolved via d93f33f

sirv@2.0 will support native ESM usage.

**Comment by lukeed** [comment]
Closed via d93f33f and #86 
Will be in next release.
